src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,208 @@
+                return null;
+            }
+
+            public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationExpression node)
+            {
+                // If the delegate creation was resolved to a single static method, we do not examine the
+                // receiver of the method group as it's completely unused.
+                if (node.Argument is BoundMethodGroup group)
+                {
+                    VerifyNode(group);
+                    if (node.MethodOpt?.IsStatic == false)
+                    {
+                        Visit(group.ReceiverOpt);
+                    }
+
+                    return null;
+                }
+                else
+                {
+                    return base.VisitDelegateCreationExpression(node);
+                }
+            }
+
+            public override BoundNode VisitNewT(BoundNewT node)
+            {
+                // https://github.com/dotnet/roslyn/issues/33387 We're not currently
+                // examining child nodes correctly
+                if (node.InitializerExpressionOpt != null)
+                {
+                    VerifyNode(node.InitializerExpressionOpt, overrideSkippedExpression: true);
+                }
+
+                return null;
+            }
+
+            public override BoundNode VisitBadExpression(BoundBadExpression node)
+            {
+                // Regardless of what the BadExpression is, we need to add all of its direct children to the visited map. They
+                // could be things like object initializers (see New_01.F1).
+                foreach (var child in node.ChildBoundNodes)
+                {
+                    if (!s_skippedExpression.Contains(child.Kind))
+                    {
+                        Visit(child);
+                    }
+                    else
+                    {
+                        VerifyNode(child, overrideSkippedExpression: true);
+                    }
+                }
+
+                return null;
+            }
+
+            public override BoundNode VisitQueryClause(BoundQueryClause node)
+            {
+                Visit(node.UnoptimizedForm ?? node.Value);
+                return null;
+            }
+
+            public override BoundNode VisitUnboundLambda(UnboundLambda node)
+            {
+                Visit(node.BindForErrorRecovery().Body);
+                return null;
+            }
+
+            public override BoundNode VisitForEachStatement(BoundForEachStatement node)
+            {
+                this.Visit(node.IterationVariableType);
+                this.Visit(node.Expression);
+                // PROTOTYPE(nullable-api): handle the deconstruction
+                //this.Visit(node.DeconstructionOpt);
+                this.Visit(node.Body);
+                return null;
+            }
+
+            public override BoundNode VisitGotoStatement(BoundGotoStatement node)
```

Comment:
>VisitGotoStatement [](start = 38, length = 18)

Taking another example: I'm not clear whether this override is needed/helpful. Since it contains no expressions, it seems harmless to let the default visit take place.